server/flow.go

```diff
@@ -0,0 +1,825 @@
+		WithText("@{{ .DelegatedTo }} completed configuring the integration.").
+		Next(stepDone)
+}
+
+func (fm *FlowManager) stepInstanceURL() flow.Step {
+	enterpriseText := "Are you using `gitlab.com`?"
+	return flow.NewStep(stepGitlabURL).
+		WithText(enterpriseText).
+		WithButton(flow.Button{
+			Name:  "Yes",
+			Color: flow.ColorDefault,
+			OnClick: func(f *flow.Flow) (flow.Name, flow.State, error) {
+				err := fm.setGitlabURL(gitlab.Gitlabdotcom)
+				if err != nil {
+					return "", nil, err
+				}
+
+				return stepOAuthInfo, nil, nil
+			},
+		}).
+		WithButton(flow.Button{
+			Name:  "No",
+			Color: flow.ColorPrimary,
+			Dialog: &model.Dialog{
+				Title:            "GitLab URL",
+				IntroductionText: "Enter the **GitLab URL** of your GitLab instance (Example: https://gitlab.example.com).",
+				SubmitLabel:      "Save & continue",
+				Elements: []model.DialogElement{
+					{
+
+						DisplayName: "GitLab URL",
+						Name:        "gitlab_url",
+						Type:        "text",
+						SubType:     "url",
+						Placeholder: "Enter GitLab URL",
+					},
+				},
+			},
+			OnDialogSubmit: fm.submitGitlabURL,
+		}).
+		WithButton(cancelButton())
+}
+
+func (fm *FlowManager) submitGitlabURL(f *flow.Flow, submitted map[string]interface{}) (flow.Name, flow.State, map[string]string, error) {
+	errorList := map[string]string{}
+
+	gitlabURLRaw, ok := submitted["gitlab_url"]
+	if !ok {
+		return "", nil, nil, errors.New("gitlab_url missing")
+	}
+	gitlabURL, ok := gitlabURLRaw.(string)
+	if !ok {
+		return "", nil, nil, errors.New("gitlab_url is not a string")
+	}
+
+	err := isValidURL(gitlabURL)
+	if err != nil {
+		errorList["gitlab_url"] = err.Error()
+	}
+
+	if len(errorList) != 0 {
+		return "", nil, errorList, nil
+	}
+
+	config := fm.getConfiguration()
+	config.GitlabURL = gitlabURL
+	config.sanitize()
+
+	configMap, err := config.ToMap()
+	if err != nil {
+		return "", nil, nil, err
+	}
+
+	err = fm.client.Configuration.SavePluginConfig(configMap)
+	if err != nil {
+		return "", nil, nil, errors.Wrap(err, "failed to save plugin config")
+	}
+
+	return "", flow.State{
+		keyGitlabURL: config.GitlabURL,
+	}, nil, nil
+}
+
+func (fm *FlowManager) stepOAuthInfo() flow.Step {
+	oauthPretext := `
+##### :white_check_mark: Step 1: Register an OAuth Application in GitLab
+You must first register the Mattermost GitLab Plugin as an authorized OAuth app.`
+	oauthMessage := fmt.Sprintf(""+
+		"1. In a browser, go to {{ .GitlabURL }}/-/profile/applications.\n"+
+		"2. Set the following values:\n"+
+		"	- Name: `Mattermost GitLab Plugin - <your company name>`\n"+
+		"	- Redirect URI: `%s/oauth/complete`\n"+
+		"3. Unselect **Expire access tokens**.\n"+
+		"4. Select `api` and `read_user` in Scopes.\n"+
+		"5. Select **Save application**\n",
+		fm.pluginURL,
+	)
+
+	return flow.NewStep(stepOAuthInfo).
+		WithPretext(oauthPretext).
+		WithText(oauthMessage).
+		//TODO: Add image
+		// WithImage(fm.pluginURL, "public/new-oauth-application.png").
```

Comment:
@Rina-dsg Could you please provide a screenshot for the "Create Application" step?

Reply:
It looks like this right now. Please let me know what you think.
![Screenshot from 2022-02-24 11-51-53](https://user-images.githubusercontent.com/16541325/155510884-39e2947a-873f-4b06-b5ab-bc4edd3a2551.png)
![Screenshot from 2022-02-24 11-51-59](https://user-images.githubusercontent.com/16541325/155510880-f381a7ad-e0a1-49e0-811a-dd1fb28fc2fa.png)